mppsolar/protocols/pi18sv.py

```diff
@@ -0,0 +1,1008 @@
+        "response": [
+            ["ack", "Command execution", {"NAK": "Failed", "ACK": "Successful"}],
+        ],
+        "test_responses": [
+            b"^1\x0b\xc2\r",
+            b"^0\x1b\xe3\r",
+        ],
+        "regex": "MUCHGC(\\d,\\d\\d\\d)$",
+    },
+    "MCHGV": {
+        "name": "MCHGV",
+        "prefix": "S015",
+        "crc": True,
+        "description": "Set battery maximum charge voltage",
+        "help": " -- MCHGVmmm,nnn with mmm battery constant charge voltage and nnn battery float voltage. Unit 0.1V",
+        "type": "SETTER",
+        "response": [
+            ["ack", "Command execution", {"NAK": "Failed", "ACK": "Successful"}],
+        ],
+        "test_responses": [
+            b"^1\x0b\xc2\r",
+            b"^0\x1b\xe3\r",
+        ],
+        "regex": "MCHGV(\\d\\d\\d,\\d\\d\\d)$",
+    },                
+    "DAT": {
+        "name": "DAT",
+        "prefix": "S018",
+        "crc": True,
+        "description": "Set date time",
+        "help": " -- examples: DAT190518224530(YYMMDDHHMMSS-12digits)",
+        "type": "SETTER",
+        "response": [
+            ["ack", "Command execution", {"NAK": "Failed", "ACK": "Successful"}],
+        ],
+        "test_responses": [
+            b"^1\x0b\xc2\r",
+            b"^0\x1b\xe3\r",
+        ],
+        "regex": "DAT(\\d\\d\\d\\d\\d\\d\\d\\d\\d\\d\\d\\d)$",
+    },
+    "POP": {
+        "name": "POP",
+        "prefix": "S007",
+        "crc": True,
+        "description": "Set output source priority",
+        "help": " -- POP0 : Solar-Utility-Battery ; POP1 : Solar-Battery-Utility",
+        "type": "SETTER",
+        "response": [
+            ["ack", "Command execution", {"NAK": "Failed", "ACK": "Successful"}],
+        ],
+        "test_responses": [
+            b"^1\x0b\xc2\r",
+            b"^0\x1b\xe3\r",
+        ],
+        "regex": "POP(\\d)$",
+    },
+    "BUCD": {
+        "name": "BUCD",
+        "prefix": "S014",
+        "crc": True,
+        "description": "Battery re-charged and re-discharged voltage when utility is available",
+        "help": " -- BUCDmmm,nnn with mmm battery recharge voltage when utility available (44 to 51V) and nnn Battery re-discharged voltage when utility is available (0,48-58)",
+        "type": "SETTER",
+        "response": [
+            ["ack", "Command execution", {"NAK": "Failed", "ACK": "Successful"}],
+        ],
+        "test_responses": [
+            b"^1\x0b\xc2\r",
+            b"^0\x1b\xe3\r",
+        ],
+        "regex": "BUCD(\\d\\d\\d,\\d\\d\\d)$",
+    },
+    
+}
+
+
+class pi18sv(AbstractProtocol):
+    def __str__(self):
+        return "PI18-SV protocol handler"
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__()
+        self._protocol_id = b"PI18SV"
+        self.COMMANDS = COMMANDS
+        self.STATUS_COMMANDS = ["PI","T","ET","EY","EM","ED","ID","VFW","PIRI","GS","MOD","FWS","FLAG","DI","CHGCR","MUCHGCR","PRI","PGS","ACCT","ACLT"]
```

Comment:
STATUS_COMMANDS are the commands to run when a --getStatus command is issued. This list looks too long (normally it is 1 or 2 or 3 key commands)